openedx/adg/lms/applications/management/commands/update_is_prerequisite_courses_passed.py

```diff
@@ -25,121 +27,64 @@ def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
         if not prereq_course_groups:
             sys.exit('Exiting!!! No open pre-req courses found but there must be some pre-reqs. Please add from admin')
 
-        users_to_be_checked_for_update = self.get_minimal_users_to_be_checked_for_update(prereq_course_groups)
+        user_ids_with_program_pre_reqs_not_marked_as_passed = self.get_users_with_pre_reqs_not_marked_as_passed()
+        users_to_be_checked_for_update = get_users_with_active_enrollments_from_course_groups(
+            user_ids_with_program_pre_reqs_not_marked_as_passed, prereq_course_groups
+        )
 
         if not users_to_be_checked_for_update:
-            logger.info('None of the users passed pre-req courses')
+            logger.info('No new user has passed pre-req courses')
             sys.exit(0)
 
-        self.check_users_for_application_update(users_to_be_checked_for_update, prereq_course_groups)
-
-    def get_minimal_users_to_be_checked_for_update(self, prereq_course_groups):
-        """
-        Filter minimal set of users who should be checked for the update.
-        Args:
-            prereq_course_groups: List of course groups
-        Returns:
-            A list of user ids who should be checked for the update
-        """
-        users_to_be_checked_for_update = self.get_users_with_pre_reqs_not_marked_as_passed()
-
-        for prereq_course_group in prereq_course_groups:
-            users_to_be_checked_for_update = self.get_users_with_active_course_enrollments(
-                users_to_be_checked_for_update, prereq_course_group
-            )
-
-        return users_to_be_checked_for_update
+        eligible_users = self.get_users_eligible_for_update(users_to_be_checked_for_update)
+        self.update_is_prerequisite_courses_passed_in_application_hub(eligible_users)
 
     def get_users_with_pre_reqs_not_marked_as_passed(self):
```

Comment:
since now we have two types of prereqs, program prereqs and business unit prereqs, lets rename this method to

`get_users_with_program_pre_reqs_not_marked_as_passed`